Source/WebCore/rendering/GridTrackSizingAlgorithm.cpp

```diff
@@ -1639,38 +1758,43 @@ void GridTrackSizingAlgorithm::resolveIntrinsicTrackSizes(GridLayoutState& gridL
 
 void GridTrackSizingAlgorithm::resolveIntrinsicTrackSizesMasonry(GridLayoutState& gridLayoutState)
 {
-    if (m_strategy->isComputingSizeContainment()) {
+    if (m_strategy->isComputingSizeContainment() || !m_grid.hasGridItems()) {
         handleInfinityGrowthLimit();
         return;
     }
+    StdMap<SpanLength, MasonryMinMaxTrackSize> indefiniteItems;
+    StdMultimap<SpanLength, MasonryMinMaxTrackSizeWithGridSpan> definiteItems;
+    Vector<MasonryMinMaxTrackSizeWithGridSpan> definiteItemsThatSpanFlexTrack;
 
-    Vector<GridTrack>& allTracks = tracks(m_direction);
-    Vector<GridItemWithSpan> itemsSortedByIncreasingSpan;
-    Vector<GridItemWithSpan> itemsCrossingFlexibleTracks;
-    SingleThreadWeakHashSet<RenderBox> itemsSet;
-    MasonryIndefiniteItems masonryIndefiniteItems;
+    // Compute Span Sizes.
+    computeIndefiniteItemsForMasonry(indefiniteItems, definiteItems, definiteItemsThatSpanFlexTrack, gridLayoutState);
 
-    if (m_grid.hasGridItems()) {
-        computeIndefiniteItemsForMasonry(masonryIndefiniteItems, gridLayoutState);
+    // Update intrinsic tracks with single span items that do not cross flex tracks.
+    auto& allTracks = tracks(m_direction);
 
+    auto item = indefiniteItems.find(1);
+    if (item != indefiniteItems.end()) {
```

Comment:
Since you don't use `item` outside of the if

```suggestion    
    if (auto item = indefiniteItems.find(1); item != indefiniteItems.end()) {
```